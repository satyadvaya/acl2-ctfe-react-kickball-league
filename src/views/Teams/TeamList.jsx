import { Link } from "react-router-dom";

function TeamList() {
  return (
    <>
      <h1>Teams:</h1>
      <ul>
        <li>
          <Link to="/teams/1" className="App-link">
            Team 1
          </Link>
        </li>
        <li>
          <Link to="/teams/2" className="App-link">
            Team 2
          </Link>
        </li>
        <li>
          <Link to="/teams/3" className="App-link">
            Team 3
          </Link>
        </li>
      </ul>
    </>
  );
}

export default TeamList;
