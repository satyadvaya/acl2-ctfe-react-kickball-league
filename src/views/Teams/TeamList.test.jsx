import { screen, render } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router";
import TeamList from "./TeamList";

it("should render the team list", async () => {
  render(
    <MemoryRouter initialEntries={["/teams/"]}>
      <Route path="/teams" component={TeamList} />
    </MemoryRouter>
  );

  const teamName = await screen.findByText("Stumptown Lumberjacks", {
    exact: false,
  });
  expect(teamName).toBeInTheDocument();
});
