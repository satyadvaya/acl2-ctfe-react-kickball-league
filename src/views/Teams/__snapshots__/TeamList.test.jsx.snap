// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the team list 1`] = `
<div>
  <h1>
    Teams:
  </h1>
  <a
    class="App-link"
    href="/teams/new"
  >
    Add a team
  </a>
  <ul>
    <li>
      <a
        class="App-link"
        href="/teams/1"
      >
        Mocked Team 1
      </a>
        
      <button
        type="button"
      >
        Update
      </button>
        
      <button
        aria-label="Delete Mocked Team 1"
        type="button"
      >
        Delete
      </button>
    </li>
    <li>
      <a
        class="App-link"
        href="/teams/2"
      >
        Mocked Team 2
      </a>
        
      <button
        type="button"
      >
        Update
      </button>
        
      <button
        aria-label="Delete Mocked Team 2"
        type="button"
      >
        Delete
      </button>
    </li>
  </ul>
</div>
`;
